### PR TITLE
Moving replayer out of lua

### PR DESCRIPTION
**Pull request by ebetica**
Doing this so we can do `#include "replayer.h"` when using libtorchcraft.

### CODE REVIEW

File: include/replayer.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ */
+
+#pragma once
+
+extern "C" {
+#include <TH/TH.h>
+#include <lua.h>
+#include <luaT.h>
+#include <lauxlib.h>
+#include <lualib.h>
```
Comment (jgehring) [inline include/replayer.h]: `TH/TH.h` should be the only one of those required, the others can be moved to `replayer_lua.h`

File: include/replayer.h [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#include <cstdio>
+
+#include "refcount.h"
+#include "frame.h"
+
+#ifdef _MSC_VER
+typedef unsigned int uint32_t;
+typedef unsigned short uint16_t;
+typedef int int32_t;
+#endif
```
Comment (jgehring) [inline include/replayer.h]: These shouldn't be necessary with VS 2012 and upwards.

File: replayer/replayer.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  int32_t key, val;
+  for (auto i = 0; i < s; i++) {
+    in >> key >> val;
+    o.numUnits[key] = val;
+  }
+
+  return in;
+}
+
+// Constructor/Destructor/Load/Save
```
Comment (jgehring) [inline replayer/replayer.cpp]: Nit: remove this?

**Comment by jgehring** [comment]
Do you want to move the frame diff change to a new PR? It would also be good to apply clang-format with the style file from #60 to this.

**Comment by ebetica** [comment]
Oops, I screwed up git branches, should be okay now. It looks like there's a lot of formatting to do, so I can do that in a separate PR.

**Pull request merged**